### DISCUSSION TITLE
When I link to the opc server, can I get the value of quailty on the opc server

### DISCUSSION THREAD

**Discussion by armandxu** [eclipse-milo/milo#1100] [General]
When I link to the opc server, can I get the value of quailty on the opc server？
<img width="668" alt="1674090505208" src="https://user-images.githubusercontent.com/7020744/213332400-53b9b96e-e741-48a5-aa82-ae893549337a.png">

**Comment by kevinherron** [comment]
When you subscribe to the Value attribute you receive a DataValue, which has as StatusCode component.

**Reply by armandxu** [reply]
hi，kevinherron，thank you for your reply , but why when the opc ua server quality stamp is bad, but the StatusCode in the DataValue received by the opc ua client is good.
<img width="651" alt="1674871185928" src="https://user-images.githubusercontent.com/7020744/215236276-244d43ca-d280-41a3-be27-a3d68e501c99.png">

<img width="875" alt="1674871124812" src="https://user-images.githubusercontent.com/7020744/215236249-eb3bb671-8755-4759-8b6d-7d674b008ddb.png">

**Comment by kimim** [comment]
As indicated in code:

```java
    private static String quality(StatusCode statusCode) {
        if (statusCode.isGood()) {
            return "good";
        } else if (statusCode.isBad()) {
            return "bad";
        } else if (statusCode.isUncertain()) {
            return "uncertain";
        } else {
            return "unknown";
        }
    }
```

quality is just a string name for status code name.

**Reply by armandxu** [reply]
hi，kimim，thanks for your reply ,Can you show the complete code for reading data quality?
